src/main/java/com/uw/model/TrainingType.java

```diff
@@ -1,24 +1,53 @@
 package com.uw.model;
 
+import jakarta.persistence.*;
+
+@Entity
 public class TrainingType {
-    private String trainingTypeName;
 
-    public TrainingType(String trainingTypeName){
-        this.trainingTypeName = trainingTypeName;
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "ID")
+    private long id;
+
+    @Enumerated(EnumType.STRING)
+    @Column(name = "TYPE_NAME")
+    private TrainingTypeEnum typeName;
+
+    public TrainingType() {
+        // no-args constructor
+    }
+
+    public TrainingType(TrainingTypeEnum typeName) {
+        this.typeName = typeName;
+    }
+
+    public TrainingTypeEnum getTypeName() {
+        return typeName;
     }
 
-    public String getTrainingTypeName() {
-        return trainingTypeName;
+    public void setTypeName(TrainingTypeEnum typeName) {
+        this.typeName = typeName;
     }
 
-    public void setTrainingTypeName(String trainingTypeName) {
-        this.trainingTypeName = trainingTypeName;
+    public void setTypeName(String typeName){
+        if (typeName.equalsIgnoreCase("YOGA")){
```

Comment:
Switch expression is less verbose . Sample:

`var result = switch(month) {
    case JANUARY, JUNE, JULY -> 3;
    case FEBRUARY, SEPTEMBER, OCTOBER, NOVEMBER, DECEMBER -> 1;
    case MARCH, MAY, APRIL, AUGUST -> 2;
    default -> 0; 
};`

Enums are allowed as parameter of the switch and does not need to use equalsIgnoreCase method

Would be nice to have it here and all switch in your code

Reply:
Thank you very much for the review and advice. I really appreciate it!